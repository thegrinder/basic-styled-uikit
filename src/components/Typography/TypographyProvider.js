import { createProvider } from '../../helpers';

const TypographyProvider = createProvider('typography');

export default TypographyProvider;
