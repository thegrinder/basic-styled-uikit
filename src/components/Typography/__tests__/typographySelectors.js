import { minimalTheme } from '../TypographyProvider';
import {
  getTypography,
  getBaseFontSize,
  getBaseLineHeight,
  rem,
  calcLineHeight,
} from '../typographySelectors';

const theme = { typography: minimalTheme };

describe('typographySelectors', () => {
  describe('getTypography', () => {
    it('should return typography theme', () => {
      expect(getTypography(theme)).toEqual(minimalTheme);
    });

    it('should throw if no theme found', () => {
      expect(() => getTypography()).toThrow();
    });

    it('should throw if no typography namespace found', () => {
      expect(() => getTypography({})).toThrow();
    });
  });

  describe('getBaseFontSize', () => {
    it('should return the correct base font size', () => {
      expect(getBaseFontSize(theme)).toEqual(theme.typography.baseFontSize);
    });
  });

  describe('getBaseLineHeight', () => {
    it('should return the correct base line height', () => {
      expect(getBaseLineHeight(theme)).toEqual(theme.typography.baseLineHeight);
    });
  });

  describe('rem', () => {
    it('should return the correct size in rem units with a default theme', () => {
      expect(rem(16)({ theme })).toEqual('1rem');
    });

    it('should return the correct size in rem units with a custom theme', () => {
      const customTheme = {
        ...theme,
        typography: {
          ...theme.typography,
          baseFontSize: 8,
        },
      };
      expect(rem(16)({ theme: customTheme })).toEqual('2rem');
    });
  });

  describe('calcLineHeight', () => {
    it('should return the correct line height according to vertical rhythm and base line height', () => {
      expect(calcLineHeight('1rem', 1.5)).toEqual('1.5');
      expect(calcLineHeight('2rem', 1.5)).toEqual('1.5');
      expect(calcLineHeight('3rem', 1.5)).toEqual('1');
    });
  });
});
