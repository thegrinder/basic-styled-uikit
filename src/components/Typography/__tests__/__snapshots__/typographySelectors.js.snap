// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography theme should be defined 1`] = `
Object {
  "baseFontSize": 16,
  "baseLineHeight": 1.5,
  "colors": Object {
    "danger": "#F4511E",
    "neutral": "#161616",
    "primary": "#1E88E5",
    "success": "#43A047",
    "warning": "#FDD835",
  },
  "heading": Object {
    "sizings": Object {
      "h1": Object {
        "fontSize": "3rem",
      },
      "h2": Object {
        "fontSize": "2.5rem",
      },
      "h3": Object {
        "fontSize": "2rem",
      },
      "h4": Object {
        "fontSize": "1.5rem",
      },
      "h5": Object {
        "fontSize": "1.25rem",
      },
      "h6": Object {
        "fontSize": "1rem",
      },
    },
  },
  "onBackground": Object {
    "danger": Object {
      "neutral": "#fff",
    },
    "primary": Object {
      "neutral": "#fff",
    },
    "success": Object {
      "neutral": "#fff",
    },
    "warning": Object {
      "neutral": "#161616",
    },
  },
  "opacities": Object {
    "high": 1,
    "low": 0.6,
    "normal": 0.8,
  },
  "text": Object {
    "sizings": Object {
      "l": Object {
        "fontSize": "1.25rem",
      },
      "m": Object {
        "fontSize": "1rem",
      },
      "s": Object {
        "fontSize": ".875rem",
      },
      "xs": Object {
        "fontSize": ".75rem",
      },
    },
  },
}
`;
